apis/addonconfigs/cni/v1alpha2/antreaconfig_types.go

```diff
@@ -288,51 +288,166 @@ type AntreaNsx struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default:=false
 	Enable bool `json:"enable,omitempty"`
-	// BootstrapFrom either providerRef or inline configs
-	// +kubebuilder:validation:Optional
-	BootstrapFrom AntreaNsxBootstrapFrom `json:"bootstrapFrom,omitempty"`
 	// Config is  configuration for nsxt adapter
 	// +kubebuilder:validation:Optional
 	AntreaNsxConfig AntreaNsxConfig `json:"config,omitempty"`
 }
 
-type AntreaNsxBootstrapFrom struct {
-	// ProviderRef is used with uTKG, which will be filled by uTKG Addon Controller
+type AntreaNsxConfig struct {
+	// BootstrapFrom either providerRef or inline configs
 	// +kubebuilder:validation:Optional
-	ProviderRef *AntreaNsxProvider `json:"providerRef,omitempty"`
-	// Inline is used with TKGm, user need to fill in manually
+	BootstrapFrom string `json:"bootstrapFrom,omitempty"`
+
+	// InfraType is the type for infrastructure, so far it is vSphere, VMC, AWS, Azure
 	// +kubebuilder:validation:Optional
-	Inline *AntreaNsxInline `json:"inline,omitempty"`
-}
+	InfraType string `json:"infraType,omitempty"`
 
-type AntreaNsxProvider struct {
-	// Api version for nsxServiceAccount, its value is "nsx.vmware.com/v1alpha1" now
 	// +kubebuilder:validation:Optional
-	ApiGroup string `json:"apigroup,omitempty"`
-	// Kind is the kind for crd, here its value is NsxServiceAccount
+	NSXCert string `json:"nsxCert,omitempty"`
+
 	// +kubebuilder:validation:Optional
-	Kind string `json:"kind,omitempty"`
-	// Name is the name for NsxServiceAccount
+	NSXKey string `json:"nsxKey,omitempty"`
+
 	// +kubebuilder:validation:Optional
-	Name string `json:"name,omitempty"`
-}
+	NSXUser string `json:"nsxUser,omitempty"`
```

Comment:
I think we don't need NSXUser and NSXPassword, maybe you forgot to remove them? They are useful for bootstrap Job, but we remove the bootstrap Job from the Package. Bootstrap Job is only used in interworking CI.

Reply:
done.